### PR TITLE
RD-5402 Fix drilldown functionality for pages defined in page groups

### PR DESCRIPTION
**Pull request by qooban**
## Description

Drilldown functionality was broken for pages inside page groups. It was assumed that `pageMenuItemReducer` receives always a page (of `PageDefinition` type) and not `PageMenuItem`.

## Screenshots / Videos

The following video shows that breadcrumbs are properly defined and pages/widgets are updated in the store:

https://user-images.githubusercontent.com/5202105/179928657-7ef240c5-7a62-4ed7-bccd-1363a29f1502.mp4

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3179/) - all PASSED

## Documentation
N/A

### CODE REVIEW

**Comment by qooban** [comment]
@kubama It's a draft, but it was tricky to detect that problem and I'm not sure the solution is the right one, so I'd like you to take a look at this stage and provide your insights as I think you've got more experience in that area.

**Comment by qooban** [comment]
@NorbertAmartus Can you re-validate RD-4703 with changes from this PR?

File: app/reducers/pageReducer.ts [hunk 1]
```diff
@@ -98,18 +109,24 @@ const pageMenuItemReducer = (state: PageDefinition, action: AnyAction) => {
         case types.REMOVE_LAYOUT_SECTION:
             return { ...state, layout: _.without(state.layout, _.nth(state.layout, action.layoutSection)) };
         case types.ADD_DRILLDOWN_PAGE: {
-            const pageData = _.cloneDeep(state);
-            forAllWidgets(pageData, layoutSectionWidgets => widgets(layoutSectionWidgets, action));
+            let pageMenuItem = _.cloneDeep(state);
+
+            // Update widget that created drilldown page
+            const pagesList: PageDefinition[] = pageMenuItem.type === 'pageGroup' ? pageMenuItem.pages : [pageMenuItem];
+            _.each(pagesList, p => forAllWidgets(p, layoutSectionWidgets => widgets(layoutSectionWidgets, action)));
```
Comment (NorbertAmartus) [inline app/reducers/pageReducer.ts:116]: WDYT about ommiting acronyms? 🤔 

```suggestion
            _.each(pagesList, page =>
                forAllWidgets(page, layoutSectionWidgets => widgets(layoutSectionWidgets, action))
            );
```

**Review by NorbertAmartus — COMMENTED** [review]
> @NorbertAmartus Can you re-validate [RD-4703](https://cloudifysource.atlassian.net/browse/RD-4703) with changes from this PR?

Sure 💪

**Comment by NorbertAmartus** [comment]
@qooban Issue seems to not appear anymore! 🎉 
So we've killed 2 birds with one stone ... in an elegant way 😁 

Because of that, I'll close #2072 with a corresponding message 🚀 
Additionally, after merging this PR, I'll mark [RD-4703](https://cloudifysource.atlassian.net/browse/RD-4703) as done ✔️

File: app/reducers/pageReducer.ts (hunk 1 above)
Comment (qooban) [reply app/reducers/pageReducer.ts:116]: Sure, will apply it.

**Review by NorbertAmartus — APPROVED** [review]

**Review by kubama — APPROVED** [review]

**Pull request merged**